vapi/simulator/simulator.go

```diff
@@ -2105,3 +2121,63 @@ func (s *handler) vmID(w http.ResponseWriter, r *http.Request) {
 func (s *handler) debugEcho(w http.ResponseWriter, r *http.Request) {
 	r.Write(w)
 }
+
+func (s *handler) consoleCLIStatus(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodGet:
+		OK(w, s.consoleCLI.Enabled)
+	case http.MethodPut:
+		var input console_cli.Access
+		if s.decode(r, w, &input) {
```

Comment:
If `s.decode` fails it returns with HTTP 400. According to the VAPI spec for these APIs they'd always err with HTTP 500. @dougm I wonder if we should enhance `decode` or add another method to influence the HTTP status code in case of an error. Otherwise we'd not comply with certain APIs and break clients if they rely on status codes, e.g. for tests, etc.

Reply:
currently, i have updated code to handle the s.decode failure
```
if s.decode(r, w,&input){
...
} else{
  w.WriteHeader(http.StatusInternalServerError)
}
```